services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -866,6 +868,76 @@ class ProtectedPathSpec extends BaseSpecification {
         database.resetCosts()
     }
 
+    def "System doesn't reroute main flow path when protected path is broken and new alt path is available\
+(altPath is more preferable than mainPath)"() {
+        given: "Two active neighboring switches with three paths at least"
+        def (Switch srcSwitch, Switch dstSwitch) = getNeighboringSwitchPair(3)
+
+        and: "A flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        and: "All alternative paths are unavailable (bring ports down on the source switch)"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        List<PathNode> broughtDownPorts = []
+        def altPaths = database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.findAll {
+            it != currentPath && it != currentProtectedPath
+        }.unique { it.first() }
+
+        altPaths.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        and: "ISL on a protected path is broken(bring port down) for changing the flow state to DOWN"
+        def protectedIslToBreak = pathHelper.getInvolvedIsls(currentProtectedPath)[0]
+        northbound.portDown(protectedIslToBreak.dstSwitch.dpId, protectedIslToBreak.dstPort)
+
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.DOWN }
+
+        when: "Make the current path less preferable than alternative path"
+        def alternativePath = altPaths.first()
+        def currentIsl = pathHelper.getInvolvedIsls(currentPath)[0]
+        def alternativeIsl = pathHelper.getInvolvedIsls(alternativePath)[0]
+
+        pathHelper.makePathMorePreferable(alternativePath, currentPath)
+
+        assert islUtils.getIslInfo(currentIsl).get().cost > islUtils.getIslInfo(alternativeIsl).get().cost
```

Comment:
Probably we can get all links once and use them to get info for a certain ISL to not get links twice.

Reply:
I'll leave it as is
otherwise it will get the following:
```def allLinks = northbound.getAllLinks()
    allLinks.find {
        it.source.switchId == currentIsl.srcSwitch.dpId && it.source.portNo == currentIsl.srcPort &&
               it.destination.switchId == currentIsl.dstSwitch.dpId && it.destination.portNo == currentIsl.dstPort
    }.cost < allLinks.find {
        it.source.switchId == alternativeIsl.srcSwitch.dpId && it.source.portNo == alternativeIsl.srcPort &&
                it.destination.switchId == alternativeIsl.dstSwitch.dpId && it.destination.portNo == alternativeIsl.dstPort
    }.cost
```

Reply:
```
def allIsls = northbound.getAllLinks()
assert islUtils.getIslInfo(allIsls, currentIsl).get().cost > islUtils.getIslInfo(allIsls, alternativeIsl).get().cost
```

But I would recommend a new way (if you rebase on the latest develop):
```
northbound.getLink(currentIsl).cost > northbound.getLink(alternativeIsl).cost
```

Reply:
looks like the `getLink` method is not merged yet into the `develop` branch